core/execution/common/liquidity_provision.go

```diff
@@ -546,8 +552,8 @@ func (m *MarketLiquidity) AmendLiquidityProvision(
 
 		proposedCommitmentVariation = pendingAmendment.CommitmentAmount.ToDecimal().Sub(lpa.CommitmentAmount.ToDecimal())
 	} else {
-		if lp != nil {
-			proposedCommitmentVariation = lp.CommitmentAmount.ToDecimal().Sub(lpa.CommitmentAmount.ToDecimal())
+		if currentProvision != nil {
```

Comment:
for all those from here to line 590, did you expect to use currentProvision or provisionToCopy?

Reply:
Current provision mate.

Reply:
cool cool